Jenkinsfile

```
@@ -25,39 +25,58 @@ pipeline {
-                          env.TAG = env.TAG_NAME ? env.TAG_NAME : DOCKER_TAGS[env.BRANCH_NAME]
-                          sh "./gradlew dockerPush"
+                    if (env.BRANCH_NAME == 'develop') {
+                        withCredentials([string(credentialsId: 'SONAR_TOKEN', variable: 'SONAR_TOKEN')]){
+                            sh(script: "./gradlew sonarqube -x test --configure-on-demand \
+                                -Dsonar.links.ci=${BUILD_URL} \
+                                -Dsonar.github.pullRequest=${env.CHANGE_ID} \
+                                -Dsonar.github.disableInlineComments=true \
+                                -Dsonar.host.url=https://sonar.soramitsu.co.jp \
+                                -Dsonar.login=${SONAR_TOKEN} \
+                                ")
                         }
                     }
                 }
             }
         }
-        stage('Sonar') {
-                  steps {
-                    script {
-                      if (env.BRANCH_NAME == 'develop') {
-                        withCredentials([string(credentialsId: 'SONAR_TOKEN', variable: 'SONAR_TOKEN')]){
-                          sh(script: "./gradlew sonarqube -x test --configure-on-demand \
-                            -Dsonar.links.ci=${BUILD_URL} \
-                            -Dsonar.github.pullRequest=${env.CHANGE_ID} \
-                            -Dsonar.github.disableInlineComments=true \
-                            -Dsonar.host.url=https://sonar.soramitsu.co.jp \
-                            -Dsonar.login=${SONAR_TOKEN} \
-                            ")
-                          }
-                      }
+        stage('Push artifacts') {
+            steps {
+                script {
+                    if (env.BRANCH_NAME ==~ /(master|develop)/) {
+                        DOCKER_TAGS = ['master': 'latest', 'develop': 'develop']
+                        withCredentials([usernamePassword(credentialsId: 'nexus-soramitsu-rw', usernameVariable: 'DOCKER_REGISTRY_USERNAME', passwordVariable: 'DOCKER_REGISTRY_PASSWORD')]) {
+                            env.DOCKER_REGISTRY_URL = "https://nexus.iroha.tech:19004"
+                            env.TAG = DOCKER_TAGS[env.BRANCH_NAME]
+                            sh "./gradlew dockerPush"
+                        }
+                        withCredentials([usernamePassword(credentialsId: 'nexus-nbc-deploy', usernameVariable: 'DOCKER_REGISTRY_USERNAME', passwordVariable: 'DOCKER_REGISTRY_PASSWORD')]) {
+                            env.DOCKER_REGISTRY_URL = "https://nexus.iroha.tech:19000"
+                            env.TAG = DOCKER_TAGS[env.BRANCH_NAME]
+                            sh "./gradlew dockerPush"
+                        }
+                    } else if (env.TAG_NAME ==~ /^(bakong-)(.*)/) {
+                        def tagPattern = (env.TAG_NAME =~ /(?<=bakong-)(\d{1,2}\.\d.*)/)[0][1]
```

it can be more then 99 version in 10 years 
```suggestion
                        def tagPattern = (env.TAG_NAME =~ /(?<=bakong-)(\d+\.\d.*)/)[0][1]
```

---

I hope so))))